### PR TITLE
chore(headless): document each specific "TestUtils"

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1410

### CODE REVIEW

File: packages/headless/src/test/mock-engine.ts [hunk 1]
```diff
@@ -60,6 +60,10 @@ export interface MockSearchEngine
   extends SearchEngine<SearchAppState>,
     MockEngine {}
 
+/**
+ * Returns non-functionnal `SearchEngine`.
+ * To be used only for unit testing controllers, not for functionnal tests.
+ */
```
Comment (ThibodeauJF) [inline packages/headless/src/test/mock-engine.ts:66]: Could also add "deprecate" if we want to remove them in the future

Comment (btaillon) [reply packages/headless/src/test/mock-engine.ts:66]: IMO, if we deprecate them I don't think we should also document them. So far I prefer the idea of deprecating all of them.

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/test/mock-raw.ts [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import {Raw} from '../api/search/search/raw';
 
+/**
+ * Returns the `Raw` property of a `Result`.
+ * @param config partial `Raw`.
+ * @returns the new `Raw`.
```
[the remark this answers is not in the document]
Comment (jpmarceau) [reply packages/headless/src/test/mock-raw.ts:6]: I tried something there, but I don't really understand what's `@param config partial 'Raw'.` though. Anything someone show know when passing this param?

**Review by jpmarceau — COMMENTED** [review]
I wasn't exactly sure whether this was for a component that would get documented. I tried generating the parsed_doc.json and it did not include these strings, so I guess not?

Anyways, I tried to review like I would normal strings :)

File: packages/headless/src/test/mock-engine.ts (hunk 1 above)
Comment (olamothe) [reply packages/headless/src/test/mock-engine.ts:66]: I would add a disclaimer: "For internal use only". 

Like making it obvious that we don't encourage people to use this...

File: packages/headless/src/test/index.ts [hunk 3]
```diff
@@ -1,4 +1,17 @@
-export * from './mock-engine';
-export * from './mock-state';
-export * from './mock-result';
-export * from './mock-raw';
+export {
```
Comment (olamothe) [inline packages/headless/src/test/index.ts:1]: That's something we should have caught on earlier I guess.

It seems we use that for Atomic, so we are guilty of the same thing we should not encourage clients to do... 

That is, treating internal stuff from a package (Headless) as something that can be seen as "public" and backward compatible.

File: packages/headless/src/test/mock-raw.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/headless/src/test/mock-raw.ts]: Raw is a property or Result which contains all the non standard fields

**Comment by ThibodeauJF** [comment]
@jpmarceau We would probably have to update the parser to gather this doc, but we don't want it made public too much, so I guess it's fine.

File: packages/headless/src/test/index.ts (hunk 3 above)
Comment (ThibodeauJF) [reply packages/headless/src/test/index.ts:1]: Glad not too much stuff was exposed in the end 😅

File: packages/headless/src/test/mock-engine.ts [hunk 4]
```diff
@@ -97,6 +115,10 @@ export interface MockProductListingEngine
   extends ProductListingEngine<ProductListingAppState>,
     MockEngine {}
 
+/**
```
Comment (jpmarceau) [inline packages/headless/src/test/mock-engine.ts:118]: Should this one be for internal use only as wel?

**Review by jpmarceau — APPROVED** [review]

**Pull request merged**